src/metrics.jl

```diff
@@ -422,13 +422,13 @@ const corr_dist = CorrDist()
 const chisq_dist = ChiSqDist()
 
 # KLDivergence
-@inline eval_op(dist::KLDivergence, ai, bi) =
-    ai > 0 ? ai * log(ai / bi) : zero(eval_op(dist, oneunit(ai), bi))
+@inline eval_op(dist::KLDivergence, ai::T, bi::T) where T =
+    ai == zero(T) ? zero(eval_op(dist, oneunit(ai), bi)) : ai * log(ai / bi)
```

Comment:
A bit simpler and without restricting the types of the arguments (which should be problematic for e.g. `kl_divergence(Float32[...], Float64[...])`?):
```suggestion
@inline eval_op(dist::KLDivergence, ai, bi) =
    iszero(ai) ? zero(eval_op(dist, oneunit(ai), bi)) : ai * log(ai / bi)
```

Distances has a tradition of trying to be very lightweight and to minimize the number of dependencies, otherwise an alternative could have been to use `LogExpFunctions.xlogy` here.

Reply:
Thanks. Now updated.